src/databricks/labs/ucx/source_code/base.py

```diff
@@ -336,3 +337,77 @@ def is_a_notebook(path: Path, content: str | None = None) -> bool:
         logger.warning(f"Could not read file {path}")
         return False
     return file_header == magic_header
+
+
+@dataclass
+class DirectFsAccess:
+    """A record describing a Direct File System Access"""
+
+    UNKNOWN = "unknown"
+
+    path: str
+    is_read: bool
+    is_write: bool
+    source_id: str = UNKNOWN
+    source_timestamp: int = -1
+    source_lineage: str = UNKNOWN
+    job_id: int = -1
+    job_name: str = UNKNOWN
+    task_key: str = UNKNOWN
+    assessment_start_timestamp: int = -1
+    assessment_end_timestamp: int = -1
```

Comment:
```suggestion
    source_timestamp: datetime.datetime
    source_lineage: list[LineageAtom]
    job_id: int = -1
    job_name: str = UNKNOWN
    task_key: str = UNKNOWN
    assessment_start_timestamp: datetime.datetime
    assessment_end_timestamp: datetime.datetime
```

see https://github.com/databrickslabs/lsql/pull/272

I'm also struggling to find where `assessment_start_timestamp` and `assessment_end_timestamp` are used. we have the `source_timestamp` already, so why do we need two more timestamps?

Reply:
because there's never enough information to analyze issues in prod. Alternately we could have the run id of the workflow that generated these DFSAs but I couldn't find how to get that information from the code being executed by the workflow